packages/server/src/app.ts

```diff
@@ -174,4 +180,15 @@ export async function shutdownApp(): Promise<void> {
   await closeDatabase();
   closeRedis();
   closeRateLimiter();
+  closeWebSockets();
+
+  if (server) {
+    server.close();
+    setImmediate(() => {
+      if (server) {
+        server.emit('close');
```

Comment:
Why does this need to be delayed from immediate execution?

Reply:
Good question, ugly answer.

The problem was that the `ws` package prevented the server from cleanly shutting down.  The `setImmediate` hack is found in a few places.  The discussion here is not super satisfying, but it works: https://stackoverflow.com/questions/14626636/how-do-i-shutdown-a-node-js-https-server-immediately

Reply:
Makes sense — would you mind adding that link as a comment alongside the code for future reference?

Reply:
Just tested again without it, and it works now.... so I just removed it 🤷